project_forecast_line_milestone/models/project_task.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2024 Therp BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    @api.depends("date_deadline", "milestone_id", "milestone_id.target_date")
+    def _compute_forecast_date_planned_end(self):
+        """Override method to use milestone_id.target_date"""
+        for this in self:
+            if this.milestone_id.target_date:
+                this.forecast_date_planned_end = this.milestone_id.target_date
+                continue
+            if this.date_deadline:
+                this.forecast_date_planned_end = this.date_deadline
+                continue
+            this.forecast_date_planned_end = this.forecast_date_planned_end
```

Comment:
This appears to be possible in 14.0

Reply:
@ntsirintanis yup seems to go quite well!